Training data: 360
Validation data: 240
3D Conv Out: (None, 50, 32, 32, 64)
3D Conv Out Reshape: (None, 32, 32, 64)
Resnet18 Out: (None, 1, 1, 512)
Resnet18 Linear Out: (None, 256)
Input to GRU: (None, 50, 256)
GRU Out: (None, 50, 29)
ResNet LSTM Pretrain weights loaded
outv: (None, 200, 256)
outa: (None, 200, 256)
hidden (None, 256)
attn_out: (None, 200, 512)
attn_states: (None, 200, 200)
fusion: (None, 200, 512)

==================================================================================================
Total params: 29,658,113
Trainable params: 18,093,569
Non-trainable params: 11,564,544
__________________________________________________________________________________________________

Model weights path: test_tdavss_Offset_BothHigh_epochs40_5lr1e-4_exp1

Epoch 1/3
 1/30 [>.............................] - ETA: 3:45 - loss: 39.5725 - snr_acc: -39.5725 2/30 [=>............................] - ETA: 2:19 - loss: 33.2856 - snr_acc: -33.2856 3/30 [==>...........................] - ETA: 1:48 - loss: 29.7003 - snr_acc: -29.7003 4/30 [===>..........................] - ETA: 1:31 - loss: 26.6580 - snr_acc: -26.6580 5/30 [====>.........................] - ETA: 1:21 - loss: 24.2615 - snr_acc: -24.2615 6/30 [=====>........................] - ETA: 1:13 - loss: 22.4876 - snr_acc: -22.4876 7/30 [======>.......................] - ETA: 1:07 - loss: 21.1053 - snr_acc: -21.1053 8/30 [=======>......................] - ETA: 1:02 - loss: 19.7238 - snr_acc: -19.7238 9/30 [========>.....................] - ETA: 57s - loss: 18.7838 - snr_acc: -18.7838 10/30 [=========>....................] - ETA: 53s - loss: 17.7683 - snr_acc: -17.768311/30 [==========>...................] - ETA: 49s - loss: 16.8070 - snr_acc: -16.807012/30 [===========>..................] - ETA: 46s - loss: 16.0868 - snr_acc: -16.086813/30 [============>.................] - ETA: 43s - loss: 15.3833 - snr_acc: -15.383314/30 [=============>................] - ETA: 40s - loss: 14.7286 - snr_acc: -14.728615/30 [==============>...............] - ETA: 37s - loss: 14.1921 - snr_acc: -14.192116/30 [===============>..............] - ETA: 34s - loss: 13.7015 - snr_acc: -13.701517/30 [================>.............] - ETA: 31s - loss: 13.4418 - snr_acc: -13.441818/30 [=================>............] - ETA: 29s - loss: 13.0541 - snr_acc: -13.054119/30 [==================>...........] - ETA: 26s - loss: 12.6656 - snr_acc: -12.665620/30 [===================>..........] - ETA: 24s - loss: 12.3120 - snr_acc: -12.312021/30 [====================>.........] - ETA: 21s - loss: 11.9627 - snr_acc: -11.962722/30 [=====================>........] - ETA: 19s - loss: 11.6178 - snr_acc: -11.617823/30 [======================>.......] - ETA: 16s - loss: 11.3058 - snr_acc: -11.305824/30 [=======================>......] - ETA: 14s - loss: 11.0185 - snr_acc: -11.018525/30 [========================>.....] - ETA: 11s - loss: 10.8173 - snr_acc: -10.817326/30 [=========================>....] - ETA: 9s - loss: 10.5794 - snr_acc: -10.5794 27/30 [==========================>...] - ETA: 7s - loss: 10.3675 - snr_acc: -10.367528/30 [===========================>..] - ETA: 4s - loss: 10.1264 - snr_acc: -10.126429/30 [============================>.] - ETA: 2s - loss: 9.9266 - snr_acc: -9.9266  Traceback (most recent call last):
  File "main_tasnet_unsync.py", line 162, in <module>
    callbacks=[reducelronplateau, checkpoint_save_weights, metrics_wandb, LoggingCallback(print_fcn=log_to_file), metrics_unsync], verbose=1)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/keras/engine/training.py", line 1297, in fit_generator
    steps_name='steps_per_epoch')
  File "/usr/local/lib/python3.6/dist-packages/wandb/keras/__init__.py", line 104, in new_generator
    return old_generator(*args, **kwargs)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/keras/engine/training_generator.py", line 332, in model_iteration
    callbacks.on_epoch_end(epoch, epoch_logs)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/keras/callbacks.py", line 298, in on_epoch_end
    callback.on_epoch_end(epoch, logs)
  File "/data/AV-speech-separation1/metrics.py", line 472, in on_epoch_end
    preds = self.model.predict(Data_predict_attention(val_folders_samp_dict))
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/keras/engine/training.py", line 909, in predict
    use_multiprocessing=use_multiprocessing)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/keras/engine/training_v2.py", line 462, in predict
    steps=steps, callbacks=callbacks, **kwargs)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/keras/engine/training_v2.py", line 444, in _model_iteration
    total_epochs=1)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/keras/engine/training_v2.py", line 123, in run_one_epoch
    batch_outs = execution_function(iterator)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/keras/engine/training_v2_utils.py", line 86, in execution_function
    distributed_function(input_fn))
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/eager/def_function.py", line 457, in __call__
    result = self._call(*args, **kwds)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/eager/def_function.py", line 503, in _call
    self._initialize(args, kwds, add_initializers_to=initializer_map)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/eager/def_function.py", line 408, in _initialize
    *args, **kwds))
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/eager/function.py", line 1848, in _get_concrete_function_internal_garbage_collected
    graph_function, _, _ = self._maybe_define_function(args, kwargs)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/eager/function.py", line 2150, in _maybe_define_function
    graph_function = self._create_graph_function(args, kwargs)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/eager/function.py", line 2041, in _create_graph_function
    capture_by_value=self._capture_by_value),
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/framework/func_graph.py", line 915, in func_graph_from_py_func
    func_outputs = python_func(*func_args, **func_kwargs)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/eager/def_function.py", line 358, in wrapped_fn
    return weak_wrapped_fn().__wrapped__(*args, **kwds)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/keras/engine/training_v2_utils.py", line 73, in distributed_function
    per_replica_function, args=(model, x, y, sample_weights))
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/distribute/distribute_lib.py", line 760, in experimental_run_v2
    return self._extended.call_for_each_replica(fn, args=args, kwargs=kwargs)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/distribute/distribute_lib.py", line 1787, in call_for_each_replica
    return self._call_for_each_replica(fn, args, kwargs)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/distribute/distribute_lib.py", line 2132, in _call_for_each_replica
    return fn(*args, **kwargs)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/autograph/impl/api.py", line 292, in wrapper
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/keras/engine/training_v2_utils.py", line 162, in _predict_on_batch
    return predict_on_batch(model, x)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/keras/engine/training_v2_utils.py", line 370, in predict_on_batch
    return model(inputs)  # pylint: disable=not-callable
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/keras/engine/base_layer.py", line 847, in __call__
    outputs = call_fn(cast_inputs, *args, **kwargs)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/keras/engine/network.py", line 708, in call
    convert_kwargs_to_constants=base_layer_utils.call_context().saving)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/keras/engine/network.py", line 860, in _run_internal_graph
    output_tensors = layer(computed_tensors, **kwargs)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/keras/engine/base_layer.py", line 847, in __call__
    outputs = call_fn(cast_inputs, *args, **kwargs)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/keras/engine/sequential.py", line 270, in call
    outputs = layer(inputs, **kwargs)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/keras/engine/base_layer.py", line 847, in __call__
    outputs = call_fn(cast_inputs, *args, **kwargs)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/keras/layers/convolutional.py", line 197, in call
    outputs = self._convolution_op(inputs, self.kernel)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/ops/nn_ops.py", line 1134, in __call__
    return self.conv_op(inp, filter)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/ops/nn_ops.py", line 639, in __call__
    return self.call(inp, filter)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/ops/nn_ops.py", line 238, in __call__
    name=self.name)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/ops/gen_nn_ops.py", line 1553, in conv3d
    dilations=dilations, name=name)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/framework/op_def_library.py", line 631, in _apply_op_helper
    param_name=input_name)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/framework/op_def_library.py", line 60, in _SatisfiesTypeConstraint
    ", ".join(dtypes.as_dtype(x).name for x in allowed_list)))
TypeError: Value passed to parameter 'input' has DataType uint8 not in list of allowed values: float16, bfloat16, float32, float64
